metron-platform/metron-test-utilities/src/main/java/org/apache/metron/test/mock/MockHTable.java

```diff
@@ -437,15 +463,14 @@ public ResultScanner getScanner(byte[] family, byte[] qualifier)
     return getScanner(scan);
   }
 
-  List<Put> putLog = new ArrayList<>();
-
   public List<Put> getPutLog() {
-    return putLog;
+    return ImmutableList.copyOf(putLog.get());
   }
 
   @Override
   public void put(Put put) throws IOException {
-    putLog.add(put);
+    putLog.get().add(put);
```

Comment:
See my response to @justinleet per `synchronizedList`.

Per your testing question: "I have triggered this code to run in Travis CI many times without seeing a failure. Prior to this the Travis CI builds would more likely fail than not."